### ISSUE TITLE
get_field_display not working as expected

### ISSUE THREAD

**Issue by willunicamp** [ra-systems/django-slick-reporting#74]
I'm using an IntegerField populated with choices, but the report and the chart are showing the integer value. I thought it was not implemented, but looking into the code of generator.py I saw that it is and I still couldn't figure out why it is not working as expected.

This is the part of the code describing the IntegerField and the columns/chart using slick reporting.

```  
   #Choices and model field
   STATUS = (
            (0,"Solicitada"),
            (1,"Agendada"),
            (2,"Concluída"),
            (3,"Cancelada"),
            (4,"Anulada"),
            )

       status = models.IntegerField("Status", choices=STATUS, default=1, null=False) 


      # slick reporting columns and chart code
      columns = ['status',
               SlickReportField.create(Count, 'id', name='monitoria__count', verbose_name='Monitorias'),
               ]


    chart_settings = [
        Chart(
            "Monitorias por status",
            Chart.PIE,
            data_source=["monitoria__count"],
            title_source=["status"],
        ),
    ]
```

**Comment by willunicamp** [comment]
Sure! 

```
from django.db.models import Count
from slick_reporting.views import ReportView, Chart
from slick_reporting.fields import  SlickReportField
from monitoria import models as monitoria_models


class MonitoriaPorStatus(ReportView):
    report_model = monitoria_models.Monitoria
    date_field = 'horario__inicio'
    group_by = 'status'

    columns = ['status',
               SlickReportField.create(Count, 'id', name='monitoria__count', verbose_name='Monitorias'),
               ]


    chart_settings = [
        Chart(
            "Monitorias por status",
            Chart.PIE,
            data_source=["monitoria__count"],
            title_source=["status"],
        ),
    ]


    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        m = monitoria_models.Monitoria._meta.get_field('status').choices
        print(m)
```
The print inside the init method is just a fast debug to be sure that choices are not empty. This is the output:

> ((0, 'Solicitada'), (1, 'Agendada'), (2, 'Concluída'), (3, 'Cancelada'), (4, 'Anulada'))

And below are the chart and the report. I was expecting the text from the choices to be shown instead of the numbers.

![image](https://github.com/ra-systems/django-slick-reporting/assets/4306213/de9cd3b4-cfd3-4097-a287-2f3187ea2251)

**Comment by RamezIssac** [comment]
Hello @willunicamp 

Actually it's indeed not implemented on ``ReportGenerator``, it's implemented on the ``ListViewReportGenerator`` which is for displaying records straight from model, like an admin changelist.

To acheive what you want, you can use the `format_row` hook on the view to adjust the display 

```python 
    def format_row(self, row_obj):
        index = row_obj['status']
        if index == 0:
            row_obj["status"] = "Solicitada"
       elif index == 1:
            row_obj["status"] = "Agendada"
       # etc
```
You can also explore Django's Choices Enum https://docs.djangoproject.com/en/4.2/ref/models/fields/#enumeration-types to make the whole choice label thing easier .


```python 
    class Monitoria(models.Model):
        class StatusChoicesEnum(models.IntChoices):
            Solicitada = 0, _("Solicitada")
            Agendada = 1, _("Agendada")
            # ........

            
    def format_row(self, row_obj):
        index = row_obj['status']
        return Monitoria.StatusChoicesEnum(index).label
```

I dont think the capabilities of having choices labels automatically gathered will be implemented for ReportGenerator. Reason is: by the time the data is gathered for display, queryset is actually a dictionary with only the needed fields, so no access to get_FOO_display.
This is for effeciency and to enhance memory consumption and speed.

**Comment by willunicamp** [comment]
Thank you for the tips. 

Solved using the code below in the format_row

```python
    def format_row(self, row_obj):
        index = row_obj['status']
        row_obj['status'] = Monitoria.StatusChoicesEnum(index).label
        return row_obj
```

**Comment by RamezIssac** [comment]
Hello , Jus thought to let you know that get_FOO_display wont be supported for the ListViewReport either starting from the released version 1.1.0 (released today)
